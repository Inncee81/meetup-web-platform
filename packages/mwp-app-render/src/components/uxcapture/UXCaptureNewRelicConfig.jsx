// @flow
import React from 'react';
import UXCaptureConfig from './UXCaptureConfig';

export const onMark = (mark: string) => {
	if (window.newrelic) {
		const now = new Date().getTime();

		// Set a marker in the trace details
		window.newrelic.addToTrace({
			name: mark,
			start: now,
			type: 'UX Capture mark',
		});

		//  Add a custom attribute to the PageView & BrowserInteraction events in Insights
		if (window.performance) {
			// `window performance.timing.navigationStart` is the event that NR uses as 'start' of page load
			const timeToMark = now - window.performance.timing.navigationStart;
			window.newrelic.setCustomAttribute(mark, timeToMark);
		}
	}
};

export const onMeasure = (measure: string) => {
	if (window.newrelic) {
		const now = new Date().getTime();

		window.newrelic.addToTrace({
			name: measure,
			start: now,
			type: 'UX Capture measure',
		});

		if (window.performance) {
			const timeToMeasure = now - window.performance.timing.navigationStart;
			window.newrelic.setCustomAttribute(measure, timeToMeasure);
		}
	}
};

export default () => <UXCaptureConfig onMark={onMark} onMeasure={onMeasure} />;
