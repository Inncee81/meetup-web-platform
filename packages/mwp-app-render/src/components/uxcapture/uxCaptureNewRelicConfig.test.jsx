import React from 'react';
import { shallow } from 'enzyme';
import UXCaptureNewRelicConfig from './UXCaptureNewRelicConfig';

const renderComponent = () => shallow(<UXCaptureNewRelicConfig />);

describe('UXCaptureNewRelicConfig', () => {
	it('renders correct markup when all props are defined', () => {
		expect(renderComponent()).toMatchSnapshot();
	});
});
