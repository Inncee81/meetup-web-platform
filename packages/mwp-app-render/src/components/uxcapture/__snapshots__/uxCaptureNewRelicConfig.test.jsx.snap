// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UXCaptureNewRelicConfig renders correct markup when all props are defined 1`] = `
<Component
  onMark={[Function]}
  onMeasure={[Function]}
/>
`;
